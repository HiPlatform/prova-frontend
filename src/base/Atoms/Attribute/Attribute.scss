@import "../../../variables";

p {
  margin: 0;
  color: $darkColor;
}