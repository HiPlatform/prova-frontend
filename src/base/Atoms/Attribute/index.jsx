import React from 'react';
import PropTypes from 'prop-types';

import './Attribute.css';

const Attribute = ({ text }) => (
  <p>{text}</p>
);

Attribute.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Attribute;
