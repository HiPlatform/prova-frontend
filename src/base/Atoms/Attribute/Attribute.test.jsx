import React from 'react';
import { shallow } from 'enzyme';
import Attribute from './index';

describe('Attribute', () => {
  const component = shallow(<Attribute />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render the text prop', () => {
    component.setProps({ text: 'foo' });
    expect(component.text()).toBe('foo');
  });
});
