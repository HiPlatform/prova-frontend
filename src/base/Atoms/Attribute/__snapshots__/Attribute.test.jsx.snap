// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribute should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Attribute />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
